### ISSUE TITLE
How to plot Cox model which has risk.table group by grouped variable with ggsurvplot()?

### ISSUE THREAD

**Issue by youyuxiansen** [kassambara/survminer#478]
Hi @kassambara .
From the description in #67
![image](https://user-images.githubusercontent.com/35478566/85113370-c246d480-b249-11ea-8ac4-99b37c04174a.png)

I'm using coxph for adjusting by age and wt.loss.I want to draw the graph which has risk.table of each level of sex.

Here is an example code of my question.
`
library("survival")
# Fit cox ph
#%%%%%%%%%%%%%
res.cox <- coxph(Surv(time, status) ~ age + sex + wt.loss, data =  lung)
res.cox

# Survival curves by sex after adjusting by age and wt.loss
#%%%%%%%%%%%%%%%%%%%%%%%%
# we construct a new data frame with two rows, 
# one for each value of sex; the other covariates are fixed to their average values 
# Create the new data  
new_df <- with(lung,
               data.frame(sex = c(1, 2), 
                          age = rep(mean(age, na.rm = TRUE), 2),
                          wt.loss = rep(mean(wt.loss, na.rm = TRUE), 2)
               )
)
new_df

# Survival curves with new data
#%%%%%%%%%%%%%%%%%%%%%%%%%%%
library(survminer)
fit <- survfit(res.cox, newdata = new_df)
ggsurvplot(fit, conf.int = TRUE, palette = "Dark2", data = lung,
           censor = FALSE, surv.median.line = "hv", risk.table = TRUE)
`

**Comment by TBradley27** [comment]
Hello, I am also having this problem at the moment

**Comment by kassambara** [comment]
Short answer: you can plot the **Cox-adjusted curves** per level of `sex`, but survminer cannot attach a *per-group number-at-risk table* to them — and that second part is a genuine open request (see #67).

Adjusted curves from your `newdata` work fine:

```r
library(survival); library(survminer)
res.cox <- coxph(Surv(time, status) ~ age + sex + wt.loss, data = lung)
new_df  <- with(lung, data.frame(sex = c(1, 2),
                                 age = mean(age, na.rm = TRUE),
                                 wt.loss = mean(wt.loss, na.rm = TRUE)))
ggsurvplot(survfit(res.cox, newdata = new_df), data = new_df,
           risk.table = TRUE, legend.labs = c("sex=1", "sex=2"))
```

![cox-adjusted curves; risk table collapses to All](https://i.imgur.com/Gr8wliH.png)

Notice the risk table shows a single **"All"** row, not one per sex: a Cox-adjusted curve is a prediction for a hypothetical average-covariate profile, so there is no real per-group at-risk count to tabulate. `ggadjustedcurves(res.cox, variable = "sex")` draws the same adjusted curves but likewise has no risk table.

If you want a true **per-sex** number-at-risk table, those counts come from the observed data, i.e. the unadjusted KM:
```r
ggsurvplot(survfit(Surv(time, status) ~ sex, data = lung), data = lung, risk.table = TRUE)
```

Combining *adjusted curves* with a *per-group risk table* in one call isn't supported today, so I'll leave this open as the tracking request for that enhancement (alongside #67).

**Comment by kassambara** [comment]
Keeping this open as the tracking request for a number-at-risk table on Cox-model-adjusted curves grouped by a covariate. (#67, the earlier related thread, is now closed.)

What works today: `ggsurvplot()` produces the risk table automatically for a *stratified* `survfit`, so if you fit strata directly you get the grouped table:

```r
library(survminer); library(survival)
fit <- survfit(Surv(time, status) ~ sex, data = lung)
ggsurvplot(fit, data = lung, risk.table = TRUE, break.time.by = 200)
```

What isn't native yet — and what this issue tracks — is the same per-group number-at-risk table underneath `ggadjustedcurves()` (Cox-model-adjusted curves), where the "groups" come from a model covariate rather than survfit strata. `ggadjustedcurves()` returns a bare ggplot with no accompanying risk table, and there's no argument to request one. Leaving open for that enhancement.

**Comment by kassambara** [comment]
For anyone landing here: the practical way to do this today is in the "Customization recipes" vignette -- "Risk table under Cox-adjusted survival curves" (#231) and "Risk table under `ggadjustedcurves()`" (#286). Both overlay a Kaplan-Meier number-at-risk table for the real grouping variable (here `sex`) onto the adjusted curves, which is the only honest at-risk count -- the adjusted curves are predictions for a fixed covariate profile and have no literal number at risk of their own.

Keeping this open as the tracker for a built-in option.

**Comment by kassambara** [comment]
This is now built in. `ggadjustedcurves()` takes a `risk.table` argument, so the Cox-adjusted curves per level of `sex` and a per-group number-at-risk table come from a single call:

```r
# remotes::install_github("kassambara/survminer")
library(survminer); library(survival)

lung$sex <- factor(lung$sex, labels = c("Male", "Female"))
cox <- coxph(Surv(time, status) ~ sex + age + wt.loss, data = lung)

ggadjustedcurves(cox, variable = "sex", data = lung, risk.table = TRUE)
```

![Cox-adjusted curves by sex with a per-group number-at-risk table](https://i.imgur.com/lWAlfv2.png)

The table is labelled "Number at risk (Kaplan-Meier, by sex)" on purpose: an adjusted curve is a prediction for a fixed covariate profile and has no literal number at risk of its own, so the honest at-risk count is the observed Kaplan-Meier count for the grouping variable. `risk.table.height`, `break.time.by` and `xlim` are available too.

If you specifically want the curves for chosen covariate **profiles** (e.g. `survfit(cox, newdata = ...)` at fixed `age`/`wt.loss`) rather than the marginal adjusted curves, the manual composition -- draw the profile curves, then set `$table` to a `ggrisktable()` built from the KM fit by `sex` -- is in the "Customization recipes" vignette ("Risk table under Cox-adjusted survival curves", #231).

Closing as resolved.
